src/axom/primal/operators/detail/intersect_impl.cpp

```diff
@@ -330,35 +330,42 @@ inline bool checkVertex(const Point2& p1,
                         bool includeBoundary,
                         double EPS)
 {
-  if (isGpeq(twoDcross(r2, p2, q1), 0.0, includeBoundary, EPS))
+  // The tests `isGpeq(twoDcross(...))` are checking the orientation
+  // of the triangle defined by its three arguments (CCW vs CW)
+  // Note: Comments in this function refer to regions
+  // in Figure 8 of the paper: https://hal.inria.fr/inria-00072100/document
+
+  /* *INDENT-OFF* */
+
+  if (isGpeq(twoDcross(r2, p2, q1), 0.0, includeBoundary, EPS))                // q1 is in {R_22, R_23, R_24, R_25}
   {
-    if (isGpeq(twoDcross(q2, r2, q1), 0.0, includeBoundary, EPS))
+    if (isGpeq(twoDcross(q2, r2, q1), 0.0, includeBoundary, EPS))                  // q1 is in {R_23, R_24} or possibly R_22
     {
-      if (isGpeq(twoDcross(p1, p2, q1), 0.0, includeBoundary, EPS))
+      if (isGpeq(twoDcross(p1, p2, q1), 0.0, includeBoundary, EPS))                    // q1 in in R_23  or possible R_22
       {
         if (isLpeq(twoDcross(p1, q2, q1), 0.0, includeBoundary, EPS))
         {
-          return true;
+          return true;                                                                       // q1 is in R_23  -- intersect
         }
         else
         {
-          return false;
+          return false;                                                                      // q1 in in R_22 -- no intersection
         }
       }
-      else
+      else                                                                             // q1 is in R_24
       {
         if (isGpeq(twoDcross(p1, p2, r1), 0.0, includeBoundary, EPS) &&
-            isGpeq(twoDcross(r2, p2, r1), 0.0, includeBoundary, EPS))
+            isLpeq(twoDcross(q1, p2, r1), 0.0, includeBoundary, EPS))
```

Comment:
The bug fix is clearly here:
```diff
- isGpeq(twoDcross(r2, p2, r1), 0.0, includeBoundary, EPS))   // Interpretation: check for CCW orientation (r2,p2,r1)
+ isLpeq(twoDcross(q1, p2, r1), 0.0, includeBoundary, EPS))   // Interpretation: check for CW orientation (q1,p2,r1)
```

Reply:
Nice find!

Reply:
The bug was here:
![image](https://user-images.githubusercontent.com/5626552/86278781-c8b04700-bb8d-11ea-83b3-8b18319979c6.png)

Reply:
The case that helped me debug this case was:
```
Triangle 1 { p: (1,1),        q: (0,0),  r: (1,0)
Triangle 2 { p: (.667, .333), q: (1,-2), r: (1,-1)
```
It is in the new 2D barycentric triangle-triangle interesection tests.